pymeasure/instruments/siglenttechnologies/siglent_spdbase.py

```diff
@@ -0,0 +1,278 @@
+from pymeasure.instruments.instrument import Instrument
+from pymeasure.instruments.validators import (strict_discrete_set,
+                                              truncated_range
+                                              )
+from enum import IntFlag
+
+log = logging.getLogger(__name__)  # https://docs.python.org/3/howto/logging.html#library-config
+log.addHandler(logging.NullHandler())
+
+
+class SystemStatusCode(IntFlag):
+    """System status enums based on ``IntFlag``
+
+    Used in conjunction with :attr:`~.system_status_code`.
+
+        ======  ======
+        Value   Enum
+        ======  ======
+        256     WAVEFORM_DISPLAY
+        64      TIMER_ENABLED
+        32      FOUR_WIRE
+        16      OUTPUT_ENABLED
+        1       CONSTANT_CURRENT
+        0       CONSTANT_VOLTAGE
+        ======  ======
+
+    """
+
+    WAVEFORM_DISPLAY = 256  # bit 8 -- waveform display enabled
+    TIMER_ENABLED = 64  # bit 6 -- timer enabled
+    FOUR_WIRE = 32  # bit 5 -- four-wire mode enabled
+    OUTPUT_ENABLED = 16  # bit 4 -- output enabled
+    CONSTANT_CURRENT = 1   # bit 0 -- constant current mode
+    CONSTANT_VOLTAGE = 0   # bit 0 -- constant voltage mode
+
+
+class SPDBase(Instrument):
+    """ The base class for Siglent SPDxxxxX instruments.
+    """
+
+    def __init__(self, adapter, **kwargs):
+        super().__init__(
+            adapter,
+            usb=dict(write_termination='\n',
+                     read_termination='\n'),
+            tcpip=dict(write_termination='\n',
+                       read_termination='\n'),
+            **kwargs
+        )
+
+    error = Instrument.measurement(
+        "SYST:ERR?",
+        """Read the error code and information of the instrument.
+
+        :type: str
+        """
+    )
+
+    fw_version = Instrument.measurement(
+        "SYST:VERS?",
+        """Read the software version of the instrument.
+
+        :type: str
+        """
+    )
+
+    system_status_code = Instrument.measurement(
+        "SYST:STAT?",
+        """Read the system status register.
+
+        :type: :class:`.SystemStatusCode`
+        """,
+        get_process=lambda v: SystemStatusCode(int(v, base=16)),
+    )
+
+    local_lockout = Instrument.setting(
+        "%s",
+        """``True`` disables local interface, ``False`` enables it.
+        """,
+        validator=strict_discrete_set,
+        values={True: "*LOCK", False: "*UNLOCK"},
+        map_values=True
+    )
+
+    save_settings = Instrument.measurement(
+        "*SAV %d",
+        """Save the current configuration in non-volatile memory.
+
+        :type: int
+        """,
+    )
+
+    recall_settings = Instrument.measurement(
+        "*RCL %d",
+        """Recall the saved configuration from non-volatile memory.
+
+        :type: int
+        """,
+    )
+
+    selected_channel = Instrument.control(
+        "INST?", "INST %s",
+        """Control the selected channel of the instrument.
+
+        :type: :class:`.SPDChannel`
```

Comment:
Is the type really a class or just an index? According to "values", I'd say its an integer.